partiql-eval/src/eval/evaluable.rs

```diff
@@ -1009,30 +1009,38 @@ impl EvalOrderBy {
                 let r = spec.expr.evaluate(&r, ctx);
 
                 match spec.spec {
-                    EvalOrderBySortSpec::AscNullsFirst => l.as_ref().cmp(r.as_ref()),
-                    EvalOrderBySortSpec::AscNullsLast => match (l.as_ref(), r.as_ref()) {
-                        (Null, Null) => Ordering::Equal,
-                        (Null, Missing) => Ordering::Less,
-                        (Missing, Missing) => Ordering::Equal,
-                        (Missing, Null) => Ordering::Greater,
-                        (Null, _) => Ordering::Greater,
-                        (Missing, _) => Ordering::Greater,
-                        (_, Null) => Ordering::Less,
-                        (_, Missing) => Ordering::Less,
-                        (l, r) => l.cmp(r),
-                    },
-                    EvalOrderBySortSpec::DescNullsFirst => match (l.as_ref(), r.as_ref()) {
-                        (Null, Null) => Ordering::Equal,
-                        (Null, Missing) => Ordering::Less,
-                        (Missing, Missing) => Ordering::Equal,
-                        (Missing, Null) => Ordering::Greater,
-                        (Null, _) => Ordering::Less,
-                        (Missing, _) => Ordering::Less,
-                        (_, Null) => Ordering::Greater,
-                        (_, Missing) => Ordering::Greater,
-                        (l, r) => r.cmp(l),
-                    },
-                    EvalOrderBySortSpec::DescNullsLast => r.as_ref().cmp(l.as_ref()),
+                    EvalOrderBySortSpec::AscNullsFirst => {
+                        NullSortedValue::<true, Value>(l.as_ref()).cmp(&NullSortedValue::<
+                            true,
+                            Value,
+                        >(
+                            r.as_ref()
+                        ))
+                    }
```

Comment:
might be more readable as:

```rust
let wrap = NullSortedValue::<true, Value>;
let (l,r) = (wrap(l), wrap(r));
l.cmp(&r)
```

Reply:
Thanks, applied the suggestion. Guess that original line was so long that the `rustfmt` made the code hard to read